codeowners, roachtest, team: rename bulk-io to disaster-recovery

celiala opened the pull request:
Reflecting `bulk-io` to `disaster-recovery` team rename in:
- github CODEOWNERS
- pkg/cmd/roachtest owners
- pkg/internal/team team
- TEAMS.yaml

Release note: None

Partially fixes: DEVINFHD-652

celiala committed:
codeowners, roachtest, team: rename bulk-io to disaster-recovery

Reflecting `bulk-io` to `disaster-recovery` team rename in:
- github CODEOWNERS
- pkg/cmd/roachtest owners
- pkg/internal/team team
- TEAMS.yaml

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88075)
<!-- Reviewable:end -->

shermanCRL commented on pkg/cmd/roachtest/registry/owners.go:
```diff
@@ -17,7 +17,7 @@ type Owner string
 const (
 	OwnerSQLExperience Owner = `sql-experience`
-	OwnerBulkIO        Owner = `bulk-io`
+	OwnerBulkIO        Owner = `disaster-recovery`
```
Change this variable name while we’re at it?

jlinder commented on pkg/internal/team/team.go:
```diff
@@ -33,7 +33,7 @@ type Team struct {
 	// them. The purpose "other" indicates a team that exists but which has no
 	// particular purpose as far as `teams` is concerned (for example, teams like
 	// the @cockroachdb/bulk-prs team which exists primarily to route, via
-	// CODEOWNERS, code reviews for the @cockroachdb/bulk-io team). This map
+	// CODEOWNERS, code reviews for the @cockroachdb/disaster-recovery team). This map
```
```suggestion
	// the @cockroachdb/kv-prs team which exists primarily to route, via
	// CODEOWNERS, code reviews for the @cockroachdb/kv team). This map
```

Making this suggestion because the bulk-io / disaster-recovery team no longer follows the pattern this comment talks about.

jlinder reviewed (commented):
One suggestion.

jlinder commented:
@celiala Note that I changed the name of the @cockroachdb/disaster-recovery team back to @cockroachdb/bulk-io until this PR merges. This is because there is automation to 1) automatically assigning PRs to the team that doesn't happen if the team doesn't exist and 2) @mention teams in issues created from failed nightly tests that they own. 

Here are two possible migration paths that would provide a minimal race condition window.

- create a new @cockroachdb/disaster-recovery team that has the same settings as @cockroachdb/bulk-io except for name
- land this PR
- backport this PR to the `release-22.2`, `release-22.1`, and `release-21.2` branches (some of the backports may require manual adjustment)

or

- prepare PRs for the `release-22.2`, `release-22.1`, and `release-21.2` branches
- merge this PR and the backport PRs at about the same time
- change the @cockroachdb/bulk-io team name to @cockroachdb/disaster-recovery as soon as the PRs are merged

celiala commented:
> @celiala Note that I changed the name of the @cockroachdb/disaster-recovery team back to https://github.com/orgs/cockroachdb/teams/bulk-io until this PR merges. This is because there is automation to 1) automatically assigning PRs to the team that doesn't happen if the team doesn't exist and 2) @mention teams in issues created from failed nightly tests that they own.

Ah -- thanks for catching! I went ahead with creating temporary new @cockroachdb/disaster-recovery team: 

- since we already renamed the team for blathers https://github.com/cockroachlabs/blathers-bot/pull/91 , we'll want the new name now for the blathers-bot. 
- having the temporary name mitigates the race condition window as we backport/merge this commit.

celiala commented:
blathers backport 22.2 22.1 21.2

celiala reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88075)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @cockroach-dev-inf, @jlinder, @shermanCRL, and @smg260)

___
*[`pkg/cmd/roachtest/registry/owners.go` line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88075#-NC7JdhZDFaNp2okS80u-r1-20:-NCJjI6D5O1HqaMCra5L:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/eae1506279b1d549299dbea87ee673e66424b3ac/pkg/cmd/roachtest/registry/owners.go#L20)):*
<details><summary><i>Previously, shermanCRL (Matt Sherman) wrote…</i></summary><blockquote>

Change this variable name while we’re at it?
</blockquote></details>

Done.
___
*[`pkg/internal/team/team.go` line 36 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88075#-NC7JdhZDFaNp2okS80v-r1-36:-NCG5YFq9OM99C6ofGZj:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/eae1506279b1d549299dbea87ee673e66424b3ac/pkg/internal/team/team.go#L36)):*
<details><summary><i>Previously, jlinder (James H. Linder) wrote…</i></summary><blockquote>

```suggestion
	// the @cockroachdb/kv-prs team which exists primarily to route, via
	// CODEOWNERS, code reviews for the @cockroachdb/kv team). This map
```

Making this suggestion because the bulk-io / disaster-recovery team no longer follows the pattern this comment talks about.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

celiala commented:
TFTRs!

bors r+
